packages/main/src/index.ts

```diff
@@ -51,14 +51,17 @@ app.on('window-all-closed', () => {
   }
 });
 
-app.once('before-quit', () => {
-  extensionLoader
+app.once('before-quit', async event => {
+  event.preventDefault();
+  await extensionLoader
     ?.stopAllExtensions()
     .then(() => {
       console.log('Stopped all extensions');
+      app.quit();
     })
     .catch(error => {
       console.log('Error stopping extensions', error);
+      app.quit();
```

Comment:
I'm thinking we should move to app.on('before-quit') and add a flag to tell if we're already stopping the extensions.
because we might still quit the app before performing the stopAll extensions here



You might do a

```
try {

} catch {

} finally {
   app.quit()
}
```